client/jetpack-connect/plans.jsx

```diff
@@ -132,50 +128,25 @@ class Plans extends Component {
 		this.props.completeFlow();
 	}
 
-	autoselectPlan( props ) {
-		const { selectedPlan, selectedPlanSlug } = props;
-
-		if ( selectedPlanSlug === PLAN_JETPACK_FREE || selectedPlanSlug === 'free' ) {
-			this.selectFreeJetpackPlan();
-			return;
-		}
-		if ( selectedPlan ) {
-			this.selectPlan( selectedPlan );
-			return;
-		}
-	}
-
-	selectFreeJetpackPlan() {
-		clearPlan();
-		this.props.recordTracksEvent( 'calypso_jpc_plans_submit_free', {
-			user: this.props.userId,
-		} );
-		mc.bumpStat( 'calypso_jpc_plan_selection', 'jetpack_free' );
-
-		if ( this.props.calypsoStartedConnection ) {
-			this.redirect( CALYPSO_REDIRECTION_PAGE );
-		} else {
-			this.redirectToWpAdmin( this.props );
-		}
-	}
-
 	selectPlan = cartItem => {
 		const checkoutPath = `/checkout/${ this.props.selectedSite.slug }`;
-		clearPlan();
-
-		if ( ! cartItem || cartItem.product_slug === PLAN_JETPACK_FREE ) {
-			return this.selectFreeJetpackPlan();
-		}
 
 		if ( cartItem.product_slug === get( this.props, 'selectedSite.plan.product_slug', null ) ) {
 			return this.redirect( CALYPSO_PLANS_PAGE );
 		}
 
-		this.props.recordTracksEvent( 'calypso_jpc_plans_submit', {
-			user: this.props.userId,
-			product_slug: cartItem.product_slug,
-		} );
-		mc.bumpStat( 'calypso_jpc_plan_selection', cartItem.product_slug );
+		if ( ! cartItem || cartItem.product_slug === PLAN_JETPACK_FREE ) {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit_free', {
+				user: this.props.userId,
+			} );
+			mc.bumpStat( 'calypso_jpc_plan_selection', 'jetpack_free' );
+		} else {
+			this.props.recordTracksEvent( 'calypso_jpc_plans_submit', {
+				user: this.props.userId,
+				product_slug: cartItem.product_slug,
+			} );
+			mc.bumpStat( 'calypso_jpc_plan_selection', cartItem.product_slug );
+		}
```

Comment:
Hopefully we can retire `calypso_jpc_plans_submit_free` and just use `calypso_jpc_plans_submit` with the slug!